File: 20220913/ompp-0.8.5/opari/tool/Test/testD.c
void foo (bool b) {
  int i;

  #pragma pomp inst begin(foo)
  i = 4;
  if (b) {
    #pragma pomp inst altend(foo)
    return;
  }

  #pragma omp parallel
  {
    #pragma omp inst begin(phase1)

    #pragma omp barrier

    #pragma omp inst end(phase1)
  }

  if (b) {
    #pragma pomp inst altend(foo)
    return;
  }

  #pragma pomp inst end(foo)
}

File: 20220913/ompp-0.8.5/opari/tool/Test/testD.f90
subroutine foo(b)
  logical b
  integer i

  !$POMP INST BEGIN(foo)
  i = 4
  if ( b ) then
    !$POMP INST ALTEND(foo)
    return
  end if

  !$OMP PARALLEL

    !$OMP INST BEGIN(phase1)

    !$OMP BARRIER

    !$OMP INST END(phase1)

  !$OMP END PARALLEL

  if ( b ) then
    !$POMP INST ALTEND(foo)
    return
  end if

  !$POMP INST END(foo)
end subroutine foo
